File: Ashutosh-Singh/2D-Array/Richest-Customer-Wealth.cpp
#include <iostream>
#include <vector>
#include <climits>
using namespace std;

class Solution {
public:
    int maximumWealth(vector<vector<int>>& accounts) {
        int n = accounts.size();
        int m = accounts[0].size();
        int maxSum = INT_MIN;
        
        for(int i = 0; i < n; i++) {
            int sum = 0;
            for(int j = 0; j < m; j++) {
                sum += accounts[i][j];
            }
            maxSum = max(maxSum, sum);
        }
        return maxSum;
    }
};

int main() {
    int n, m;
    cout << "Enter the number of customers: ";
    cin >> n;
    cout << "Enter the number of accounts each customer has: ";
    cin >> m;
    
    vector<vector<int>> accounts(n, vector<int>(m));
    
    cout << "Enter the wealth for each customer in each account: \n";
    for(int i = 0; i < n; i++) {
        for(int j = 0; j < m; j++) {
            cin >> accounts[i][j];
        }
    }
    
    Solution solution;
    int result = solution.maximumWealth(accounts);
    
    cout << "The maximum wealth is: " << result << endl;
    
    return 0;
}

File: [Manya]/Leetcode/2D-Array/Richest_customer_wealth.java
import java.util.Scanner;

class Solution {
    public int maximumWealth(int[][] accounts) {
        int ans = Integer.MIN_VALUE;

        for(int i = 0; i < accounts.length; i++){
            int sum = 0;
            for(int j = 0; j < accounts[i].length; j++){
                sum = sum + accounts[i][j];
            }

            if(sum > ans){
                ans = sum;
            }
        }

        return ans;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Enter the number of customers: ");
        int m = scanner.nextInt();
        
        System.out.print("Enter the number of banks: ");
        int n = scanner.nextInt();
        
        int[][] accounts = new int[m][n];
        
        System.out.println("Enter the account balances:");
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                accounts[i][j] = scanner.nextInt();
            }
        }
        
        Solution solution = new Solution();
        int result = solution.maximumWealth(accounts);
        
        System.out.println("Maximum wealth is: " + result);
        
        scanner.close();
    }
}
